### ISSUE TITLE
e2e tests frequently fail to login via login/OAuth dialog opened by Circuit SDK 

### ISSUE THREAD

**Issue by bahnson** [circuit/circuit-sdk#69]
### Summary
The automated e2e-Tests of our app frequently fail during the login/OAuth dialog via Circuit SDK.
After entering email, clicking next, entering password, and clicking sign in, the dialog switches back to the email field, instead of showing the "Allow access" button. 

![Circuit login problems in e2e-Tests](https://user-images.githubusercontent.com/15957560/111450671-e8ce6900-8710-11eb-8907-6cdbae3b48fe.jpg)

### Details
The behaviour was investigated by running the same e2e test spec in repeated runs, each running approx. 20 sec. and performing a Circuit login before test actions, and a Circuit logout afterwards.
After several successful tests (approx. 15 tests), there are a number of subsequent tests that fail to login. After that, succeeding and failing tests alternate, with fewer succeeding tests.  In 100 repeated e2e runs, 55 runs fail during login.
 
Retrying email and password input is not  a workaround, the login dialog will cycle between email and password prompts, then. Closing and re-opening the Circuit login dialog, starting login from scratch, is also not working around the problem. 

So far, our best explanation is that we're struggling with some sort of DOS-prevention mechanism in the Circuit backend. At least, it seems reasonable for Circuit to have such a mechanism and it seems reasonable that our test setup triggers it.

We'd really appreciate if we could  exchange ideas on what is causing the observed behaviour.

**Comment by rogeru** [comment]
Yes, Circuit has rate limiting. To be sure that's the issue, please provide some client logs. The network tab of the dev tools should show you the error as well. Or an alternative would be to provide me the client ID and date/time (in a private conversation) or your E2E run and I can try to get the Circuit server logs.
